File: Leetcode_Python/mergeIntervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:

        intervals.sort(key = lambda pair: pair[0])
        result = [intervals[0]]

        for start,end in intervals[1:]:
            lastIntervalEnd = result[-1][1]

            if start <= lastIntervalEnd:
                result[-1][1] = max(lastIntervalEnd, end)
            else:
                result.append([start,end])
            
        return result

File: Leetcode_Java/mergeIntervals.java
import java.util.*;

class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));

        List<int[]> result = new ArrayList<>();
        result.add(intervals[0]);

        for (int i = 1; i < intervals.length; i++) {
            int[] lastInterval = result.get(result.size() - 1);
            int[] current = intervals[i];

            if (current[0] <= lastInterval[1]) {
                lastInterval[1] = Math.max(lastInterval[1], current[1]);
            } else {
                result.add(current);
            }
        }

        return result.toArray(new int[result.size()][]);
    }
}